packages/preview/src/browser/preview-handler.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (C) 2017 TypeFox and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { inject, injectable, named } from "inversify";
+import URI from "@theia/core/lib/common/uri";
+import { ContributionProvider } from "@theia/core";
+
+export const PreviewHandler = Symbol('PreviewHandler');
+
+export interface PreviewHandler {
+    canHandle(uri: URI): boolean;
```

Comment:
Should not it be number based, similar to `OpenHandler` to prioritize several previewers for the same URI?
You can use `Prioritizeable.prioritizeAllSync` to look up a handler, like here: https://github.com/theia-ide/theia/blob/e6902835cfa83895b81b8417c0b5de8731280761/packages/core/src/browser/label-provider.ts#L113-L118

Reply:
sounds good!